packages/integrations/src/unifi-controller/unifi-controller-integration.ts

```diff
@@ -0,0 +1,274 @@
+
+    const requestHeaders: Record<string, string> = { 
+      "Content-Type": "application/json",
+      ...this.headers,
+    };
+    if (this.csrfToken) {
+      requestHeaders["X-CSRF-TOKEN"] = this.csrfToken;
+    } 
+    logger.debug("requestHeaders: "+JSON.stringify(requestHeaders)); 
+    
+    // do stats call
+    const statsResponse = await fetchWithTrustedCertificatesAsync(requestUrl,  {
+      method: 'GET', 
+      headers: {
+        ...requestHeaders,
+      },
+      })
+      .catch((err: TypeError) => {
+        const detailMessage = String(err.cause);
+        throw new IntegrationTestConnectionError("invalidUrl", detailMessage);
+      });
+      
+    
+    
+    logger.debug("back from stats call");
+    if (!statsResponse.ok) {
+      logger.debug("status after stats call: "+statsResponse.status);
+      logger.debug("status text after stats call: "+statsResponse.statusText);
+      logger.debug("url from stats call: "+statsResponse.url);
+
+      throwErrorByStatusCode(statsResponse.status);
+    }
+
+    const result = unifiSummaryResponseSchema.safeParse(await statsResponse.json());
+    logger.debug("look what we got: "+JSON.stringify(result));
+
+    const summary: NetworkControllerSummary = {
+      wanStatus: "disabled",
+
+      wwwStatus:  "disabled",
+      wwwLatency: -1,
+      wwwPing:    -1,
+      wwwUptime:  -1,
+      
+      wifiStatus: "disabled",
+      wifiUsers:  -1,
+      wifiGuests: -1,
+
+      lanStatus: "disabled",
+      lanUsers:  -1,
+      lanGuests: -1,
+
+      vpnStatus: "disabled",
+      vpnUsers: -1,
+    };
+
+    if (result.success) {
+      for (const site of result.data.data) {
+        if (site.name !== "default") {
+          break;
+        }
+
+        for (const health of site.health) {
+          switch(health.subsystem) {
+            case "wan":
+              if (health.status === "ok") {summary.wanStatus = "enabled"}
+              break;
+            
+            case "www":
+              if (health.status === "ok") {summary.wwwStatus = "enabled"}
+              summary.wwwLatency = health.latency;
+              summary.wwwPing    = health.speedtest_ping;
+              summary.wwwUptime  = health.uptime;
+              break;
+
+            case "wlan":
+              if (health.status === "ok") {summary.wifiStatus = "enabled"}
+              summary.wifiUsers  = health.num_user;
+              summary.wifiGuests = health.num_guest;
+              break;
+
+            case "lan":
+              if (health.status === "ok") {summary.lanStatus = "enabled"}
+              summary.lanUsers  = health.num_user;
+              summary.lanGuests = health.num_guest;
+              break;
+
+            case "vpn":
+              if (health.status === "ok") {summary.vpnStatus = "enabled"}
+              summary.vpnUsers  = health.remote_user_num_active;
+              break;
+
+            default:
+              break;
```

Comment:
Empty default branch is a bad idea; we'll never have an idea when it's triggered. Either...
- throw an error when this should not be executed
- remove the undesired options from the object and not use this branch